client/cpp/encoder.cc

```diff
@@ -53,7 +54,7 @@ static void CheckValidProbability(float prob, const char* var_name) {
   if (prob < 0.0f || prob > 1.0f) {
     log("%s should be between 0.0 and 1.0 inclusive (got %.2f)", var_name,
         prob);
-    assert(false);
+    throw std::out_of_range("prob");
   }
```

Comment:
OK, reverted back to using assert(false). I thought that was only an internal thing, but now I see the external style guide also says not to use exceptions. Oh, well.